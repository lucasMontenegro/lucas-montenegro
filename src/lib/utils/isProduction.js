import globals from "lib/utils/globals"
export default function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
    // WebPack will replace this last line with a boolean
  )
}