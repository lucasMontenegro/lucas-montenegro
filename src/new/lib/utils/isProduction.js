import global from "global"
export default function isProduction () {
  return (
    global.process ? global.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}