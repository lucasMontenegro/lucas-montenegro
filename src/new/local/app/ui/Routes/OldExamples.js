export default function OldExamples () {
  return null
}