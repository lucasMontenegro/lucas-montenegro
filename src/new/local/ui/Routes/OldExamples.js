import React from "react"
import { Switch } from "react-router-dom"
import throwPropTypeErrors from "new/local/ui/__examples__/throwPropTypeErrors"
export default function OldExamples () {
  return (
    <Switch>
      {throwPropTypeErrors}
    </Switch>
  )
}