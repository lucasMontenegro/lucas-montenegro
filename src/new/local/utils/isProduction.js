const { default: globals } = require("new/local/utils/globals")
exports.default = function isProduction () {
  return (
    globals.process ? globals.process.env.NODE_ENV === `production` :
    process.env.NODE_ENV === `production`
  )
}