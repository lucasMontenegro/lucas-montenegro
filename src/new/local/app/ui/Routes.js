import React from "react"
import { BrowserRouter, Switch, Route } from "react-router-dom"
import App from "new/local/app/ui/App"
import Examples from "new/local/paperbase/Examples"
import isProduction from "new/local/utils/isProduction"
export default function Routes () {
  return (
    <BrowserRouter>
      <Switch>
        <Route component={App} />
        {isProduction() || <Route path="/examples" component={Examples} />}
      </Switch>
    </BrowserRouter>
  )
}