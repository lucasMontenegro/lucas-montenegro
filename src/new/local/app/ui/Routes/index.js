import React from "react"
import { BrowserRouter, Switch, Route } from "react-router-dom"
import isProduction from "new/local/utils/isProduction"
import App from "new/local/app/ui/App"
import Examples from "new/local/app/ui/Routes/Examples"
import OldExamples from "new/local/app/ui/Routes/OldExamples"
export default function Routes () {
  return (
    <BrowserRouter>
      {
        isProduction() ? <Route component={App} /> :
        (
          <Switch>
            <Route path="/examples" component={Examples} />
            <Route path="/old-examples" component={OldExamples} />
            <Route component={App} />
          </Switch>
        )
      }
    </BrowserRouter>
  )
}