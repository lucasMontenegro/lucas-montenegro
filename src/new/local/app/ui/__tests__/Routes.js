import React from "react"
import renderer from "react-test-renderer"
import isProduction from "new/local/utils/isProduction"
import Routes from "new/local/app/ui/Routes"
jest.mock(`react-router-dom`, () => ({
  __esModule: true,
  BrowserRouter: ({ children }) => <div className="BrowserRouter">{children}</div>,
  Switch: ({ children }) => <div className="Switch">{children}</div>,
  Route: ({ path, component }) => (
    <div className="Route">
      <div className="path">{path}</div>
      <div className="component">{component}</div>
    </div>
  ),
}))
jest.mock(`new/local/app/ui/App`, () => ({
  __esModule: true,
  default: `new/local/app/ui/App`,
}))
jest.mock(`new/local/paperbase/Examples`, () => ({
  __esModule: true,
  default: `new/local/paperbase/Examples`,
}))
jest.mock(`new/local/utils/isProduction`, () => ({
  __esModule: true,
  default: jest.fn(),
}))
describe(`new/local/app/ui/Routes`, () => {
  it(`should utilize the dependency APIs correctly`, () => {
    expect(jestUtils.getDependencies([`react`, `react-router-dom`])).toMatchSnapshot()
  })
  test.each([[true], [false]])(`should render (isProduction %j)`, isProductionValue => {
    isProduction.mockReturnValueOnce(isProductionValue)
    expect(renderer.create(<Routes />).toJSON()).toMatchSnapshot()
  })
})