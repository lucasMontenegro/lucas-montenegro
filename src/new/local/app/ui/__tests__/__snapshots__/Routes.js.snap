// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new/local/app/ui/Routes should render (isProduction false) 1`] = `
<div
  className="BrowserRouter"
>
  <div
    className="Switch"
  >
    <div
      className="Route"
    >
      <div
        className="path"
      />
      <div
        className="component"
      >
        new/local/app/ui/App
      </div>
    </div>
    <div
      className="Route"
    >
      <div
        className="path"
      >
        /examples
      </div>
      <div
        className="component"
      >
        new/local/app/ui/Examples
      </div>
    </div>
  </div>
</div>
`;

exports[`new/local/app/ui/Routes should render (isProduction true) 1`] = `
<div
  className="BrowserRouter"
>
  <div
    className="Switch"
  >
    <div
      className="Route"
    >
      <div
        className="path"
      />
      <div
        className="component"
      >
        new/local/app/ui/App
      </div>
    </div>
  </div>
</div>
`;
